### PR TITLE
Update Details Page for Threads

### PR DESCRIPTION
**Pull request by grgia**
### Details

Adds Leave Thread Button for threads. 

| Chat DM Thread | Workspace Room Thread |
| ----- | ----- |
| image | image|
| image | image |
| image | image |
| image | image | 

### Fixed Issues

$ https://github.com/Expensify/App/issues/18757

### Tests

- [ ] Verify that no errors appear in the JS console
- [ ] Open a threaded report, open details, verify `Members` and `Leave thread` options show. 
- [ ] Click `Leave Thread`. Verify that thread is removed from LHN and user is sent to concierge chat.
- [ ] Open a NON threaded report room, open details, verify `Leave room` option shows. 

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](hxttps://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

image

</details>

<details>
<summary>Mobile Web - Safari</summary>

image

</details>

<details>
<summary>Desktop</summary>

image

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

image

</details>

### CODE REVIEW

File: src/languages/es.js [hunk 1]
```diff
@@ -119,6 +119,7 @@ export default {
         enterManually: 'Introducir manualmente',
         message: 'Chatear con ',
         leaveRoom: 'Salir de la sala de chat',
+        leaveThread: 'Salir del hilo',
```
Comment (grgia) [inline src/languages/es.js:122]: Need to check this !

File: src/pages/ReportDetailsPage.js [hunk 2]
```diff
@@ -87,12 +87,24 @@ class ReportDetailsPage extends Component {
...
             menuItems.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.LEAVE_ROOM,
                 translationKey: 'common.leaveRoom',
                 icon: Expensicons.Exit,
-                action: () => Policy.leaveRoom(this.props.report.reportID),
+                action: () => Report.leaveRoom(this.props.report.reportID),
+            });
+        }
+
+        if (this.props.report.parentReportActionID) {
```
Comment (grgia) [inline src/pages/ReportDetailsPage.js:102]: once https://github.com/Expensify/App/pull/18522 is merged we can replace with `ReportUtils.isThread(report)` for readibility.

File: src/languages/es.js (hunk 1 above)
Comment (grgia) [reply src/languages/es.js:122]: https://expensify.slack.com/archives/C21FRDWCV/p1683804474846109

Comment (grgia) [reply src/languages/es.js:122]: Confirmed!

**Comment by grgia** [comment]
@chiragsalian @stitesExpensify could you help to define the QA for leaving threads?

**Comment by grgia** [comment]
Also ready for review!

**Review by stitesExpensify — DISMISSED** [review]

**Comment by grgia** [comment]
@fedirjh potentially may be fixed with this PR? https://github.com/Expensify/App/pull/19056

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Mentioned some bugs 1:1 to georgia to tackle.

**Comment by grgia** [comment]
@fedirjh I confirmed that bug is happening on main when you leave a public room

**Comment by chiragsalian** [comment]
Bugs to tackle,
- We should show settings button since controlling notifications is important
- In details tab -> settings, for a thread report don't show reportName (or make it non editable, whichever is easier)
- LeaveRoom does not work for non-policy chats. I think the backend does not allow for it. So my suggestion is to remove it completely here until the backend can be updated to work properly for this flow.

**Comment by grgia** [comment]
> 
> Sure, 
> For EC3.
> - i think the threads feature is fine as is. Nothing more needs to done wrt details page as a priority imo.
> 
> For v1/post EC3.
> - All threads should show settings tab since its important to set their notifications over there.
> - All threads should have the option to leave thread
> 
> 
> My suggestion, to tackle this in phases, i.e., work on what you can get out the door
> 1. The details tab looks great visually. I would say have one PR having share code, members and settings. In settings hide room name if its a thread for now. Remove leave room if its a thread because i think its better to get the feature out working completely vs not working properly in some places which could get confusing to users.
> 2. Then once backend updates LeaveRoom then add it to the details page via another newDot PR.

> So should we remove all the leaveRoom stuff from this PR and only have share code / members
>  it should also have settings and room name should be hidden if its a thread.

cc @chiragsalian posting this here for easy reference

**Comment by chiragsalian** [comment]
I've added CP to staging label since it would be nice if we can get this to staging early tomorrow, then later in the day it would go to production via regular App deploy. This is because i just realized we are doing DBSEC3 and may not deploy on thursday-friday. So if we get this out the door tomorrow it would be really nice. We can also ping QA to test.

**Comment by chiragsalian** [comment]
I edited out my comment about leaveRoom [here](https://github.com/Expensify/App/pull/18772#issuecomment-1550217586) since we made the backend change and its live.

**Comment by grgia** [comment]
Adding a comment to track while I'm currently updating this PR:

We will also be updating the QR details as part of this 
image

**Comment by grgia** [comment]
| Chat DM Thread | Workspace Room Thread |
| ----- | ----- |
| image | image|
| image | image |
| image | image |
| image | image |

**Comment by grgia** [comment]
@fedirjh open for review!

**Review by stitesExpensify — DISMISSED** [review]
LGTM!

**Review by fedirjh — COMMENTED** [review]
Looks good , Just noticed when the thread is initially created , the members menu item is not displayed, It get displayed after the second user open the thread.

Screenshot 2023-05-17 at 5 29 19 PM

**Comment by grgia** [comment]
@fedirjh done! Also, I think that bug has to do with our optimistic data more so than the details page, so I think we should fix it separately

**Comment by chiragsalian** [comment]
Also members page header says "Details", it should instead say "Members"

**Comment by grgia** [comment]
@fedirjh @chiragsalian addressed those bugs

**Review by stitesExpensify — APPROVED** [review]

**Review by chiragsalian — APPROVED** [review]
LGTM and it seems to work well for me

**Comment by grgia** [comment]
@fedirjh that's a known bug https://expensify.slack.com/archives/C049HHMV9SM/p1684222206427449

**Review by fedirjh — APPROVED** [review]
Thanks @grgia . Looks good to me and tests well

**Comment by Santhosh-Sellavel** [comment]
I feel we could have captured this [regression](https://github.com/Expensify/App/issues/20197) here, not an easy one though. Refer [here](https://github.com/Expensify/App/issues/20197#issuecomment-1577829745), I know the code was not added here, but we could've caught this thanks!

**Pull request merged**